src/queue/job/v1/task_sync.rs

```diff
@@ -0,0 +1,53 @@
+use crate::{
+    entity::*,
+    queue::job::{EnqueueJob, Job, JobError, SharedJobState, V1},
+};
+use sea_orm::{ColumnTrait, ConnectionTrait, EntityTrait, PaginatorTrait, QueryFilter};
+use serde::{Deserialize, Serialize};
+use time::Duration;
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq, Copy)]
+pub struct TaskSync;
+
+const SYNC_PERIOD: Duration = Duration::weeks(1);
+
+impl TaskSync {
+    pub async fn perform(
+        &mut self,
+        job_state: &SharedJobState,
+        db: &impl ConnectionTrait,
+    ) -> Result<Option<EnqueueJob>, JobError> {
+        let aggregators = Aggregators::find()
+            .filter(AggregatorColumn::IsFirstParty.eq(true)) // eventually we may want to check for a capability
+            .all(db)
```

Comment:
eventually there will be too many records to do this with and we will want to use `stream`